storm-core/src/jvm/backtype/storm/validation/ConfigValidationAnnotations.java

```diff
@@ -0,0 +1,216 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package backtype.storm.validation;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Note: every annotation interface must have method validatorClass()
+ * For every annotation there must validator class to do the validation
+ * To add another annotation for config validation, add another annotation @interface class.  Implement the corresponding
+ * validator logic in a class in ConfigValidation.  Make sure validateField method in ConfigValidation knows how to use the validator
+ * and which method definition/parameters to pass in based on what fields are in the annotation.
+ */
+public class ConfigValidationAnnotations {
+    /**
+     * Field names for annotations
+     */
+
+    static final String VALIDATOR_CLASS = "validatorClass";
+    static final String TYPE = "type";
+    static final String ENTRY_VALIDATOR_CLASSES = "entryValidatorClasses";
+    static final String KEY_VALIDATOR_CLASSES = "keyValidatorClasses";
+    static final String VALUE_VALIDATOR_CLASSES = "valueValidatorClasses";
+    static final String KEY_TYPE = "keyType";
+    static final String VALUE_TYPE = "valueType";
+    static final String INCLUDE_ZERO = "includeZero";
+
+    /**
+     * Validators with fields: validatorClass and type
+     */
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.FIELD)
+    public @interface isType {
+        Class validatorClass() default ConfigValidation.SimpleTypeValidator.class;
+
+        Class type();
+    }
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.FIELD)
+    public @interface isStrings {
```

Comment:
Can we make this areStrings or isListOfStrings.  isStrings is too close to isString for me, and it feels a bit grammatically suspect.